src/scm/bitbucket.js

```diff
@@ -1,4 +1,305 @@
+            $.get(src.links.self.href, { access_token: this.accessToken })
+            .then(content => {
+              resolve({ file: src.path, content: content});
+            })
+            .fail(reject)
+          });
+        });
+        return Promise.all(promises);
+      });
+    });
+  }
+
+  getNamespaces() {
+    return this.getAccessToken()
+    .then(accessToken => {
+      return getAllItems(Promise.resolve(
+        {
+          token: accessToken,
+          items: [], 
+          url: `${this.baseUrl}/teams?access_token=${accessToken}&role=contributor`
+        }),
+        this.followPaginate,
+        'bitbucket'
+      );
+    })
+    .then(teams => {
+      this.namespaces = [this.user].concat(teams.map(team => team.username));
+      return this.namespaces;
+    })
+    .catch((err) => {
+      showAlert('Failed to get user info.', LEVEL_ERROR);
+    });
+  }
+
+  getRepos() {
+    return this.getAccessToken()
+    .then(accessToken => {
+      return getAllItems(Promise.resolve(
+        {
+          token: accessToken,
+          items: [], 
+          url: `${this.baseUrl}/repositories/?access_token=${accessToken}&q=scm="git"&role=contributor`
+        }),
+        this.followPaginate,
+        'bitbucket'
+      );
+    })
+    .then(response => {
+      const repos = response.map(repo => repo.full_name);
+      //if current bind still existed, use it
+      const repo = context.bindRepo[context.id];
+      if (repo && $.inArray(repo.fullName, repos) >= 0) {
+        context.repo = repo;
+      }
+      return repos;
+    });
+  }
+
+  createRepo() {
+    const owner = $('#new-repo-owner').val();
+    const name = $('#new-repo-name').val();
+    const desc = $('#new-repo-desc').val();
+    const isPrivate = $('#new-repo-type').val() !== 'public';
+    const payload = {
+      scm: 'git',
+      description : desc,
+      is_private: isPrivate
+    }
+    if (!name || name === '') return;
+    return this.getAccessToken()
+    .then(() => {
+      return $.ajax({
+        url: `${this.baseUrl}/repositories/${owner}/${name}`,
+        headers: {
+          'Authorization': `Bearer ${this.accessToken}`
+        },
+        method: 'POST',
+        crossDomain: true,
+        dataType: 'json',
+        contentType: 'application/json',
+        data: JSON.stringify(payload)
+      })
+    })
+    .then(response => {
+      const repo = {
+        fullName : response.full_name
+      };
+      context.repo = repo;
+      Object.assign(context.bindRepo, { [context.id] : repo });
+      if (context.bindBranch[context.id]) {
+        delete context.bindBranch[context.id];
+      }
+      chrome.storage.sync.set({ bindRepo: context.bindRepo });
+      return response.full_name;
+    })
+    .then(repo => {
+      return this.commitFiles(repo, 'master', null, [{name: "README.md", content: "initialed by gas-github"}], null, 'initial commit')
+      .then(() => {
+        return repo;
+      });
+    })
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/leonhartX/gas-github/blob/79a1121575f894385b70c00ba30d867da9fde2de/src/scm/bitbucket.js#L235"></span>‘err’ is defined but never used.</td>
    </tr>
  </tbody>
</table>